### PR TITLE
Update Release Notes for GUI (unfinished!)

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #630.

+  Change doc pages headings

### CODE REVIEW

File: docs/manuals/gui/extension/dynamic_element/scalar_props.md [hunk 1]
```diff
@@ -129,7 +129,7 @@ In the Markdown definition of the page, we add a button:
 <|Add a character|button|id=addChar|>
 ```
 
-And implement the *on_action* callback to react to the user pressing this button:
+And implement the `on_action` callback to react to the user pressing this button:
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/extension/dynamic_element/scalar_props.md:132]: on_action with () ?

File: docs/manuals/gui/viselements/index.md [hunk 2]
```diff
@@ -280,11 +278,11 @@ can use it in your CSS selectors. You can look at the [Styling](../styling/index
 section for more information.
 
 !!! note
-    This identifier is also sent to the *on_action* callback if this visual
+    This identifier is also sent to the `on_action` callback if this visual
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/viselements/index.md:281]: () ?

File: docs/manuals/gui/viselements/index.md [hunk 3]
```diff
@@ -316,11 +314,11 @@ of the variable that holds that dictionary as the value of the `properties` prop
     <|{show_dialog}|dialog|properties=dialog_props|>
     ```
 
-### The `propagate` property
+## The `propagate` property
 
 There are situations where you don't want a variable bound to a control value (such
 as the knob location of a slider) to be updated immediately when the user manipulates
-the control. You may for example want to check the received value in the *on_change*
+the control. You may for example want to check the received value in the `on_change`
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/viselements/index.md:321]: () ?

File: docs/manuals/gui/viselements/slider.md_template [hunk 4]
```diff
@@ -118,13 +119,13 @@ Then only those values are accessible by the user:
...
 
-### Multi selection {data-source="gui:doc/examples/controls/slider-multiple.py"}
+## Multi selection {data-source="gui:doc/examples/controls/slider-multiple.py"}
 
 You can use a slider control to display multiple values and let users select each.<br/>
 To achieve that, the [*value*](#p-value) property must be initially set to an array containing
 the initial values to reflect. The slider will have one knob for each value.<br/>
-When the user moves any of the knobs, the [*on_change*](../../callbacks.md#variable-value-change)
-is invoked with the variable value set to an array containing the new selection.
+When the user moves any of the knobs, the [`on_change`](../../callbacks.md#variable-value-change)
```
Comment (FredLL-Avaiga) [inline docs/manuals/gui/viselements/slider.md_template:127]: ()?

File: docs/relnotes.md [hunk 5]
```diff
@@ -87,6 +97,29 @@ Not published yet.
 <h6 style="font-size: 1.2em"><strong><code>taipy-core</code></strong></h6>
 3.0.0
 
+- :warning: The *action* parameter of the `on_action` callback was removed for every control.<br/>
+  The signature of all *on_action8 callbacks is now unified to the following:
```
Comment (FredLL-Avaiga) [inline docs/relnotes.md]: "star on_action 8" instead of "star on_action() start"

**Review by FredLL-Avaiga — APPROVED** [review]
🍟

File: docs/manuals/gui/extension/dynamic_element/scalar_props.md (hunk 1 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/extension/dynamic_element/scalar_props.md:132]: Yeah.
There's a semantical difference between
- the callback, from the Taipy stand point (on_change, on_action, ...). There are noted in \` `code_style` \`.
- the callback **function**. This is where the callback get bound to some code.
  Then we document these \* *in_italics()* \*, with the parenthesis.

File: docs/manuals/gui/viselements/index.md (hunk 2 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/viselements/index.md:281]: Same comment as above: this is the callback notion from Taipy GUI's stand point.

File: docs/manuals/gui/viselements/index.md (hunk 3 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/viselements/index.md:321]: Same

File: docs/manuals/gui/viselements/slider.md_template (hunk 4 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/viselements/slider.md_template:127]: Same

File: docs/relnotes.md (hunk 5 above)
Comment (FabienLelaquais) [reply docs/relnotes.md]: Thanks. Missing Shift, AND parenthesis.

File: docs/migration.md [hunk 6]
```diff
@@ -7,6 +7,13 @@ hide:
 This documentation page lists the migration paths of Taipy releases as they
 were published.
 
+# From 2.x to 3.0
```
Comment (jrobinAV) [inline docs/migration.md:10]: 👍🏻

File: mkdocs.yml_template
Comment (jrobinAV) [inline mkdocs.yml_template:1]: 👍🏻

File: tools/postprocess.py [hunk 7]
```diff
@@ -141,13 +141,15 @@ def on_post_build(env):
...
                         file_was_changed = True
+                    #"""
                     # Collapse doubled <h1>/<h2> page titles
                     REPEATED_H1_H2 = re.compile(
                         r"<h1>(.*?)</h1>\s*<h2\s+(id=\".*?\")>\1(<a\s+class=\"headerlink\".*?</a>)?</h2>", re.M | re.S
                     )
                     html_content, n_changes = REPEATED_H1_H2.subn('<h1 \\2>\\1\\3</h1>', html_content)
                     if n_changes != 0:
                         file_was_changed = True
+                    #"""
```
Comment (jrobinAV) [inline tools/postprocess.py:152]: ?

File: tools/postprocess.py [hunk 8]
```diff
@@ -141,13 +141,15 @@ def on_post_build(env):
                             + f"<h1{match.group(1)}>{match.group(2)}</h1>"
                             + html_content[match.end():])
                         file_was_changed = True
+                    #"""
```
Comment (jrobinAV) [inline tools/postprocess.py:144]: ?

**Review by jrobinAV — APPROVED** [review]
Great!

File: docs/manuals/gui/callbacks.md [hunk 9]
```diff
@@ -157,12 +155,12 @@ def on_action(state, id, action):
...
 ```
 
-Line 5 shows the invocation of `invoke_long_callback()^` which does all the hard work. All the parameters
-to *heavy_function()* should be sent to the call, encapsulated in an array.<br/>
-When the call is performed, *heavy_function()* is invoked, but the application keeps running anyway:
-the user function is running in a background thread. 
+Line 5 shows the invocation of `invoke_long_callback()^` which does all the hard work. All the
+parameters to *heavy_function()* should be sent to the call, encapsulated in an array.<br/>
+When the call is performed, *heavy_function()* is invoked, but the application keeps running
+anyway: the user function is running in a background thread.
```
Comment (dinhlongviolin1) [inline docs/manuals/gui/callbacks.md]: I might replace the semicolon with 'as' or a period.

**Review by dinhlongviolin1 — APPROVED** [review]

File: tools/postprocess.py (hunk 7 above)
Comment (FabienLelaquais) [reply tools/postprocess.py:152]: Just a big fat comment to keep this code safe.

File: tools/postprocess.py (hunk 8 above)
Comment (FabienLelaquais) [reply tools/postprocess.py:144]: Just a big fat comment to keep this code safe.

File: docs/manuals/gui/callbacks.md (hunk 9 above)
Comment (FabienLelaquais) [reply docs/manuals/gui/callbacks.md]: I'll do that, thx!

**Pull request merged**